v2/protocol/http/protocol.go

```diff
@@ -116,6 +130,14 @@ func New(opts ...Option) (*Protocol, error) {
 		p.ShutdownTimeout = DefaultShutdownTimeout
 	}
 
+	if p.ReadTimeout == 0 {
```

Comment:
nit: your comment says if 0 or negative, but this only asserts on 0

Reply:
@embano1 good catch, will fix

Reply:
See commentary below